bindings/java/src/main/native/bridge.cc

```diff
@@ -51,33 +71,74 @@ static void print_error(const char* error_msg) {
     fprintf(stderr, "[ERROR] JUCX - %s: %s\n", __FILE__, error_msg);
 }
 
-static ucs_status_t create_context() {
-    if (cached_ctx) {
-        return UCS_OK;
+JNIEXPORT jlong JNICALL
+Java_org_ucx_jucx_Bridge_createWorkerNative(JNIEnv *env, jclass cls,
+        jint max_comp, jobject comp_queue, jobject jworker) {
+    ucp_worker_params_t worker_params = { 0 };
+    worker* worker_ptr = NULL;
+    ucs_status_t status;
+    jobject jbyte_buff;
+    uint32_t cap = (uint32_t) max_comp;
+    ucp_address_t* local_addr;
+    size_t local_addr_len;
+    jbyteArray jaddr_arr;
+    jlong* addr_ptr;
+    jbyte* local_addr_wrap;
+
+    worker_params.field_mask    = UCP_WORKER_PARAM_FIELD_THREAD_MODE;
+    worker_params.thread_mode   = UCS_THREAD_MODE_SINGLE;
+
+    worker_ptr = new worker(&cached_ctx, cap);
+    status = worker_ptr->init(worker_params);
+    if (status != UCS_OK) {
+        ERR_JUMP("Failed to initialize ucp native worker", err_worker);
     }
 
-    ucp_params_t ucp_params = { 0 };
-    ucp_config_t *config;
-    ucs_status_t status;
-    ucp_context_h ucp_context;
+    status = worker_ptr->extract_worker_address(&local_addr, local_addr_len);
+    if (!local_addr) {
+        ERR_JUMP("Failed to get ucp worker native address", err_worker);
+    }
 
-    status = ucp_config_read(NULL, NULL, &config);
-    if (status != UCS_OK) {
-        return status;
+    local_addr_wrap = new jbyte[local_addr_len];
+    if (!local_addr_wrap) {
+        ERR_JUMP("Allocation failure", err_worker);
     }
+    memcpy(local_addr_wrap, local_addr, local_addr_len);
 
-    ucp_params.features     = UCP_FEATURE_TAG;
-    ucp_params.field_mask   = UCP_PARAM_FIELD_FEATURES      |
-                              UCP_PARAM_FIELD_REQUEST_INIT  |
-                              UCP_PARAM_FIELD_REQUEST_SIZE;
+    jaddr_arr = env->NewByteArray(local_addr_len);
+    if (!jaddr_arr) {
+        ERR_JUMP("Failed to create Java byte[] object", err_worker_addr);
+    }
 
-    ucp_params.request_size = sizeof(jucx_request);
-    ucp_params.request_init = request_util::request_handler::request_init;
+    env->SetByteArrayRegion(jaddr_arr, 0, local_addr_len, local_addr_wrap);
+    delete[] local_addr_wrap;
+    worker_ptr->release_worker_address(local_addr);
 
-    status = ucp_init(&ucp_params, config, &ucp_context);
-    ucp_config_release(config);
+    // Set the Java workerAddress field
+    env->SetObjectField(jworker, field_worker_addr_arr, jaddr_arr);
+
+    jbyte_buff = env->NewDirectByteBuffer(worker_ptr->get_event_queue(), cap);
+    if (!jbyte_buff) {
+        env->ExceptionClear();
+        ERR_JUMP("Failed to create Java ByteBuffer object", err_worker);
```

Comment:
looks like  jaddr_arr is not released in case of error

Reply:
jaddr_arr, after [this call](https://github.com/openucx/ucx/pull/2121/files#diff-2defdf3f6a1e1c02edf25c7544524134R108), is a Java object, so the Java GC will release it.